Source/Utils/WarpXMovingWindow.cpp

```diff
@@ -204,13 +209,15 @@ WarpX::MoveWindow (bool move_j)
 
 void
 WarpX::shiftMF (MultiFab& mf, const Geometry& geom, int num_shift, int dir,
-                amrex::Real external_field)
+                bool ext_init, bool B_flag)
```

Comment:
Could we get rid of `B_flag` here too? We could either pass `str_By_ext_grid_function` etc. as arguments, or simply use a more explicit name, e.g. a struct `FieldType::E` or `FieldType::B`. These are just suggestions.

Reply:
Thnak you for the suggestion. I have now stored the field_parsers and use the stored variable in the shiftMF function.